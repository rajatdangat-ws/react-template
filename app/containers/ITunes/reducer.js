/*
 *
 * ITunes reducer
 *
 */
import produce from 'immer';
import get from 'lodash/get';
import { createActions } from 'reduxsauce';

export const initialState = { songs: [], error: null };

export const { Types: iTunesTypes, Creators: iTunesCreators } = createActions({
  requestGetSongs: ['searchTerm'],
  successGetSongs: ['data'],
  errorGetSongs: ['error'],
  clearSongs: null
});

export const iTunesReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case iTunesTypes.SUCCESS_GET_SONGS:
        draft.songs = action.data;
        break;
      case iTunesTypes.ERROR_GET_SONGS:
        draft.error = get(action.error, 'message', 'something_went_wrong');
        break;
      case iTunesTypes.CLEAR_SONGS:
        draft.searchTerm = null;
        draft.error = null;
        draft.songs = [];
        break;
      default:
        return state;
    }
  });

export default iTunesReducer;
