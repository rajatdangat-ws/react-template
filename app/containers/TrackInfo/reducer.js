/*
 *
 * TrackInfo reducer
 *
 */
import produce from 'immer';
import get from 'lodash/get';
import { createActions } from 'reduxsauce';

export const initialState = { songDetails: {}, error: null };

export const { Types: trackInfoTypes, Creators: trackInfoCreators } = createActions({
  requestGetSongDetails: ['songId', 'songDetails'],
  successGetSongDetails: ['data'],
  errorGetSongDetails: ['error'],
  clearSongDetails: {}
});

/* eslint-disable default-case, no-param-reassign */
export const trackInfoReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case trackInfoTypes.SUCCESS_GET_SONG_DETAILS:
        draft.songDetails = action.data;
        break;
      case trackInfoTypes.ERROR_GET_SONG_DETAILS:
        draft.error = get(action.error, 'message', 'something_went_wrong');
        break;
      case trackInfoTypes.CLEAR_SONG_DETAILS:
        draft.error = null;
        draft.songDetails = {};
        break;
      default:
        return state;
    }
  });

export default trackInfoReducer;
