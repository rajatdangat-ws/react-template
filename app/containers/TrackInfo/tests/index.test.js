/**
 *
 * Tests for TrackInfo
 *
 *
 */

import React from 'react';
import { renderProvider, timeout } from '@utils/testUtils';
// import { fireEvent } from '@testing-library/dom'
import { TrackInfoTest as TrackInfo, mapDispatchToProps } from '../index';
import { trackInfoTypes } from '../reducer';

const songDetails = {
  trackName: 'test track 1',
  trackId: 1,
  coverImgUrl: 'image url 1',
  artistName: 'test artist 1',
  previewUrl: 'preview url 1',
  detailsUrl: 'details url 1',
  primaryGenreName: 'hard rock',
  releaseDate: '2006-11-07T12:00:00Z',
  releaseYear: '2006',
  trackViewUrl: 'track view url'
};

describe('<TrackInfo /> container tests', () => {
  // let submitSpy

  beforeEach(() => {
    // submitSpy = jest.fn()
  });
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(
      <TrackInfo dispatchGetSongDetails={() => {}} songInfo={songDetails} dispatchClearSongDetails={() => {}} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should call dispatchClearSongDetails when the component unmounts', async () => {
    const dispatchClearSongDetails = jest.fn();
    const { unmount } = renderProvider(
      <TrackInfo
        dispatchGetSongDetails={() => {}}
        songInfo={songDetails}
        dispatchClearSongDetails={dispatchClearSongDetails}
      />
    );
    unmount();

    await timeout(500);
    expect(dispatchClearSongDetails).toHaveBeenCalled();
  });

  it('should validate mapDispatchToProps actions', async () => {
    const dispatch = jest.fn();
    const songId = 1234567;
    const actions = {
      dispatchGetSongDetails: { songId, songDetails: undefined, type: trackInfoTypes.REQUEST_GET_SONG_DETAILS },
      dispatchClearSongDetails: { type: trackInfoTypes.CLEAR_SONG_DETAILS }
    };

    const props = mapDispatchToProps(dispatch);
    props.dispatchGetSongDetails(songId);
    expect(dispatch).toHaveBeenCalledWith(actions.dispatchGetSongDetails);

    await timeout(500);
    props.dispatchClearSongDetails();
    expect(dispatch).toHaveBeenCalledWith(actions.dispatchClearSongDetails);
  });
});
