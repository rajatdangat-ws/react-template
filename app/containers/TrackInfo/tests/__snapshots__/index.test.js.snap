// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackInfo /> container tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div>
      <div
        class="TrackInfo__TrackInfoContainer-sc-15fksvm-0 dfA-Dmp"
      >
        <img
          alt="test track 1"
          class="TrackInfo__Artwork-sc-15fksvm-1 faRTqr"
          height="250px"
          width="250px"
        />
        <h2
          class="TrackInfo__TrackTitle-sc-15fksvm-2 bgzCKO"
        >
          test track 1
        </h2>
        <a
          class="TrackInfo__TrackArtist-sc-15fksvm-3 fQrehn"
          href="artist view url"
          rel="noreferrer noopener"
          target="__blank"
        >
          test artist 1
        </a>
        <p
          class="TrackInfo__GenreName-sc-15fksvm-5 jHPtqy"
        >
          hard rock
           Â· 
          2006
        </p>
        <div
          class="TrackInfo__ButtonsContainer-sc-15fksvm-4 xgcWQ"
        >
          <a
            class="ant-btn ant-btn-primary ant-btn-round"
            href="track view url"
            rel="noreferrer noopener"
            target="__blank"
          >
            <span
              aria-label="caret-right"
              class="anticon anticon-caret-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="caret-right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
                />
              </svg>
            </span>
            <p
              class="T__StyledText-gjlic1-0 kYcyRq TrackInfo__ButtonText-sc-15fksvm-6 kTyeME"
              data-testid="t"
            >
              Play
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
